lib/content-services/src/lib/search/components/search-datetime-range/search-datetime-range.component.ts

```diff
@@ -133,18 +135,43 @@ export class SearchDatetimeRangeComponent implements SearchWidget, OnInit {
 
         this.setFromMaxDatetime();
         this.enableChangeUpdate = this.settings?.allowUpdateOnChange ?? true;
+        this.context.populateFilters
+            .asObservable()
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((filtersQueries) => {
+                if (filtersQueries[this.id]) {
+                    const start = parseISO(filtersQueries[this.id].start);
+                    const end = parseISO(filtersQueries[this.id].end);
+                    this.form.patchValue({ from: start, to: end });
+                    this.form.markAsDirty();
+                    this.apply({ from: start, to: end }, true, false);
+                } else {
+                    this.reset();
+                }
+                this.context.filterLoaded.next();
+            });
+    }
+
+    ngOnDestroy() {
+        this.destroy$.next();
+        this.destroy$.complete();
     }
 
-    apply(model: Partial<{ from: Date; to: Date }>, isValidValue: boolean) {
+    apply(model: Partial<{ from: Date; to: Date }>, isValidValue: boolean, updateContext = true) {
         if (isValidValue && this.id && this.context && this.settings && this.settings.field) {
             this.isActive = true;
 
             const start = DateFnsUtils.utcToLocal(startOfMinute(model.from)).toISOString();
             const end = DateFnsUtils.utcToLocal(endOfMinute(model.to)).toISOString();
 
             this.context.queryFragments[this.id] = `${this.settings.field}:['${start}' TO '${end}']`;
+            this.context.filterRawParams[this.id] = this.context.filterRawParams[this.id] ?? {};
```

Comment:
I think you can store that part: this.context.filterRawParams[this.id] in some const too to avoid repeating getting the same object multiple times below. Even when stored in const then assigning to fields should work through reference